src/main/java/com/cloudant/client/api/CloudantClient.java

```diff
@@ -388,24 +403,19 @@ public String serverVersion() {
      * <p><b>Note</b>: The response must be closed after use to release the connection.
      *
      * @param request The HTTP request to execute.
-     * @return {@link HttpResponse}
+     * @return {@link InputStream} with response from {@link HttpConnection}
      */
-    public HttpResponse executeRequest(HttpRequestBase request) {
+    public InputStream executeRequest(HttpConnection request) {
 
-        HttpResponse response = client.executeRequest(request);
+        InputStream response = client.executeRequest(request);
         return response;
 
     }
 
-
-    /**
-     * Shuts down the connection manager used by this client instance.
-     */
-    public void shutdown() {

```

Comment:
I think we should leave the `shutdown()` method in place even if it is currently a no-op. It is already part of the API and it preserves it for future use.